mobile/mobile.go

```diff
@@ -0,0 +1,492 @@
+package mobile
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"regexp"
+	"runtime/debug"
+	"strings"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/golang/protobuf/proto"
+	"github.com/lightninglabs/faraday/frdrpc"
+	"github.com/lightninglabs/lightning-node-connect/core"
+	"github.com/lightninglabs/lightning-node-connect/mailbox"
+	"github.com/lightninglabs/loop/looprpc"
+	"github.com/lightninglabs/pool/poolrpc"
+	"github.com/lightningnetwork/lnd/build"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/autopilotrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/chainrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/invoicesrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/signrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/verrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/watchtowerrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/wtclientrpc"
+	"github.com/lightningnetwork/lnd/signal"
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+	"gopkg.in/macaroon-bakery.v2/bakery/checkers"
+	"gopkg.in/macaroon.v2"
+)
+
+type mobileClient struct {
+	lndConn *grpc.ClientConn
+
+	statusChecker func() mailbox.ClientStatus
+
+	mac *macaroon.Macaroon
+
+	registry map[string]func(context.Context, *grpc.ClientConn,
+		string, func(string, error))
+}
+
+type stubPackageRegistration func(map[string]func(context.Context,
+	*grpc.ClientConn, string, func(string, error)))
+
+type NativeCallback interface {
+	SendResult(json string)
+}
+
+var (
+	registrations = []stubPackageRegistration{
+		lnrpc.RegisterLightningJSONCallbacks,
+		lnrpc.RegisterStateJSONCallbacks,
+		autopilotrpc.RegisterAutopilotJSONCallbacks,
+		chainrpc.RegisterChainNotifierJSONCallbacks,
+		invoicesrpc.RegisterInvoicesJSONCallbacks,
+		routerrpc.RegisterRouterJSONCallbacks,
+		signrpc.RegisterSignerJSONCallbacks,
+		verrpc.RegisterVersionerJSONCallbacks,
+		walletrpc.RegisterWalletKitJSONCallbacks,
+		watchtowerrpc.RegisterWatchtowerJSONCallbacks,
+		wtclientrpc.RegisterWatchtowerClientJSONCallbacks,
+		looprpc.RegisterSwapClientJSONCallbacks,
+		poolrpc.RegisterTraderJSONCallbacks,
+		frdrpc.RegisterFaradayServerJSONCallbacks,
+	}
+
+	perms = core.GetAllMethodPermissions()
+
+	jsonCBRegex = regexp.MustCompile("(\\w+)\\.(\\w+)\\.(\\w+)")
+
+	m             *mobileClient
```

Comment:
Is there a way that we return an instance of the mobile client instead of keeping a global instance here? So that `InitLNC()` returns such an instance? Or can there only ever be one?
Not sure how the API is going to be used, so might be a stupid question.